baseapp/options.go

```diff
@@ -269,6 +264,12 @@ func (app *BaseApp) SetMempool(mempool mempool.Mempool) {
 	app.mempool = mempool
 }
 
+// SetProcessProposal sets the process proposal function for the BaseApp.
 func (app *BaseApp) SetProcessProposal(handler sdk.ProcessProposalHandler) {
 	app.processProposal = handler
 }
+
+// SetPrepareProposal sets the prepare proposal function for the BaseApp.
+func (app *BaseApp) SetPrepareProposal(handler sdk.PrepareProposalHandler) {
+	app.prepareProposal = handler
```

Comment:
are we expecting this value during runtime? if not, then this should panic if the value is already set

Reply:
Good point because this could create non-determinism.  We'll add a sealed check.